### ISSUE TITLE
Troubleshooting, Configuring LDAP Authentication for SMRT Link Login page is broken

### ISSUE THREAD

**Issue by IndyMark** [PacificBiosciences/SMRT-Link#69]
I'm attempting to configure LDAP authentication, but if I click on the link on the Troubleshooting page, it just takes me to the main wiki page.

I am trying to figure out how to translate this working ldapsearch command into the various fields on the LDAP configuration page:

```
ldapsearch -LLL -H ldap://servername001.dow.com:389 -D abcdefg@DOW.COM -w 12345678 -b 'OU=Employees,OU=Dow Users,DC=dow,DC=com' "(uid=abcdefg)"
```

**Comment by jkaralius** [comment]
Hi @IndyMark 

I think the following should get you there:

User Store Manager Class: `org.wso2.carbon.user.core.ldap.ReadOnlyLDAPUserStoreManager`
Domain Name: `dow.com`
Connection URL: `ldap://servername001.dow.com:389`
Connection Name: `CN=abcdefg,OU=Employees,OU=Dow Users,DC=dow,DC=com`
Connection Password: `12345678`
User Search Base: `OU=Employees,OU=Dow Users,DC=dow,DC=com`
Username Attribute: `uid`
User Search Filter: `(&(objectClass=person)(uid=?))`
User List Filter: `(objectClass=person)`
Display name attribute: `uid`

Additional configuration settings:

* Uncheck Read Groups
* Delete values for configuring Group Search, including:
   * Group Search Base
   * Group Name Attribute
   * Group Search Filter
   * Group List Filter
   * Role DN Pattern
   * Membership Attribute
   * Member Of Attribute

Leave the rest of the values unchanged.

**Comment by IndyMark** [comment]
Thanks!  I got it working.

Quick follow-up question – I have discovered that when I search for users to enable them in SMRT Link, it seems to only let me search by the user’s last name, even though I am using sAMAccountName in the LDAP configuration for the Username Attribute, User Search Filter, and Display Name Attribute.  Any ideas what setting I should tweak?


Mark Frank
HPC Research Scientist
Dow AgroSciences
